### ISSUE TITLE
ctm Properties 'matchBlocks' and 'connectTiles' don't seem to work with all blocks/tiles.

### ISSUE THREAD

**Issue by pmackni** [sp614x/optifine#4822]
I'm playing on MC 1.16.2 tested with both Optifine 1.16.2_HD_U_G3_Pre3, and 1.16.2_HD_U_G3 using the Conquest RP for MC 1.15 (currently latest version).

I'm slightly modifying the RP to suit my own tastes as well as attempting to fix some ctm issues instead of waiting for another update that could potentially address them, and I'm running into issues with the overlay properties.

Using the .properties file below for displaying a grass overlay on adjacent blocks, not all blocks set in `matchBlocks=` appear to function correctly. Either I'm misunderstanding what blocks are being matched by `matchBlocks`, or there's an bug within Optifine that isn't referencing all blocks/tiles correctly. The overlay (again, using below) displays on cobblestone, but not stone or several other blocks I've tested. The RP comes with several other blocks (namely logs, dirt, gravel, and sand) that all display the overlay, but all other blocks that I've tested (aside from cobblestone) have so far not displayed the overlay. When I change `matchBlocks` to `matchTiles`, everything works great (except now I have overlays on stairs, which raises some other issues not related to this one and likely need a separate submission if that's the workaround I'm forced to take).

```
matchBlocks=cobblestone stone
#'minecraft:cobblestone minecraft:stone' produces the same results, though I haven't tried using block ids yet.
method=overlay
tiles=1-17
connect=block
connectTiles=grass_top
tintIndex=0
tintBlock=minecraft:grass_block
layer=translucent
innerSeams=true
weight=111
```

Similarly (at least apparently so on the end user side), `connectTiles` does not recognize when I set it equal to 'grass_side', despite 'grass_top' above working perfectly. There is a tile in the assets/minecraft/textures/blocks folder with the name 'grass_side.png', as well as another with 'grass_block_side.png,' neither of which trigger the connected textures on blocks (with `matchBlocks`) or tiles (with `matchTiles`) listed in that file. Instead, I have to use `connectBlocks=grass_block` and add the line `faces=sides` to produce the desired result of the side overlay.

These may better belong in two separate issues, but I feel they're more than likely caused by the same root/logic/reference issue within their respective locations within the code, both failing to reference blocks/tiles as expected. That, or again I'm not fully understanding what matchBlocks and connectTiles are referencing despite reading through the documentation (though there was at least one other issue posted regarding the 'grass_side' not working as expected).

Let me know thoughts, or if this is already in the works and/or addressed in a more recent OptiFine version.

**Comment by pmackni** [comment]
So, I'd like to apologize, but I finally got some help on the Discord that led me down the right path, and I can confirm the `matchBlocks` issue was an info disconnect on my part. I've resolved that on my own and have that fully working now.

I'd like to say the same for the issue with `connectTiles` because I tried using 'grass_block_side' again on this machine with 1.16.2 U_G3_pre3 and it worked. But I know I tried this on my other computer and got different results, but now I'm wondering if I didn't change something else, too, than cause it to not work. I'll have to investigate later, and if I can't confirm that this is a bug with 1.16.2 U_G3 (not a preview), then I'll come back and close this issue as resolved.
